extensions/mssql/src/objectExplorerNodeProvider/webhdfs.ts

```diff
@@ -0,0 +1,717 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as util from 'util';
+import * as url from 'url';
+import * as fs from 'fs';
+import * as querystring from 'querystring';
+import * as request from 'request';
+import * as BufferStreamReader from 'buffer-stream-reader';
+import * as nls from 'vscode-nls';
+import { IHdfsOptions, IRequestParams } from './fileSources';
+
+const localize = nls.loadMessageBundle();
+
+export class WebHDFS {
+	private _requestParams: IRequestParams;
+	private _opts: IHdfsOptions;
+	private _url: any;
+
+	private static ErrorMessageInvalidDataStructure =
+		localize('webhdfs.invalidDataStructure', 'Invalid Data Structure');
+
+	constructor(opts: IHdfsOptions, requestParams: IRequestParams) {
+		if (!(this instanceof WebHDFS)) {
+			return new WebHDFS(opts, requestParams);
+		}
+
+		let missingProps = ['host', 'port', 'path']
+			.filter(p => !opts.hasOwnProperty(p));
+		if (missingProps && missingProps.length > 0) {
+			throw new Error(
+				util.format(
+					localize('webhdfs.missingProperties',
```

Comment:
Localized strings can't use things like %s as far as I know. The pattern would be to use 
```suggestion
					localize('webhdfs.missingProperties', 'Unable to create WebHDFS client: missing options: ${0}', missingProps.join(', '))
```

Reply:
fixed.